oob-messages/README.md

```diff
@@ -0,0 +1,86 @@
+  "body": {
+    "goal_code": "issue-vc",
+    "goal": "To issue a Faber College Graduate credential",
+    "accept": [
+      "didcomm/v2",
+      "didcomm/aip2;env=rfc587"
+    ]
+  },
+  "attachments": [
+    {
+        "id": "request-0",
+        "media_type": "application/json",
+        "data": {
+            "json": "<json of protocol message>"
+        }
+    }
+  ]
+}
+```
+The required fields for and out of band message in didcomm messaging are
+
+**type** The header conveying the DIDComm MTURI.
+**id** This value MUST be used as the parent thread ID (pthid) for the response message that follows. This may feel counter-intuitive — why not it in the thid of the response instead? The answer is that putting it in pthid enables multiple, independent interactions (threads) to be triggered from a single out-of-band invitation.
+**from** for OOB usage. The DID representing the sender to be used by recipients for future interactions. 
+more of the field can be found [here](https://identity.foundation/didcomm-messaging/spec/#discover-features-protocol-20)
+
+The URL format is as follows
+```https://<domain>/<path>?_oob=<encodedplaintextjwm>```
+where ```encodedplaintextjwm = b64urlencode(<plaintextjwm>)```, the steps in creating an out of band invitation involves. 
+
+1. Trimming of the widespaces from the plaintextjwm. e.g 
+   ```{"type":"https://didcomm.org/out-of-band/2.0/invitation","id":"69212a3a-d068-4f9d-a2dd-4741bca89af3","from":"did:example:alice","body":{"goal_code":"","goal":""},"attachments":[{"id":"request-0","media_type":"application/json","data":{"json":"<json of protocol message>"}}]}```
+
+2. Encoding it to a 64 base encoding e.g ```eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9vdXQtb2YtYmFuZC8yLjAvaW52aXRhdGlvbiIsImlkIjoiNjkyMTJhM2EtZDA2OC00ZjlkLWEyZGQtNDc0MWJjYTg5YWYzIiwiZnJvbSI6ImRpZDpleGFtcGxlOmFsaWNlIiwiYm9keSI6eyJnb2FsX2NvZGUiOiIiLCJnb2FsIjoiIn0sImF0dGFjaG1lbnRzIjpbeyJpZCI6InJlcXVlc3QtMCIsIm1lZGlhX3R5cGUiOiJhcHBsaWNhdGlvbi9qc29uIiwiZGF0YSI6eyJqc29uIjoiPGpzb24gb2YgcHJvdG9jb2wgbWVzc2FnZT4ifX1dfQ```
+
+3. Generating the inviation Url follows the following pattern ```https://<domain>/<path>?_oob=<encodedplaintextjwm>```.<br><domain> and <path> should be kept as short as possible, and the full URL should return human readable instructions when loaded in a browser. This is intended to aid new users. The _oob query parameter is required and is reserved to contain the DIDComm message string. Additional path elements or query parameters are allowed, and can be leveraged to provide coupons or other promise of payment for new users. e.g
+```
+   https://example.com/path?_oob=eyJ0eXBlIjoiaHR0cHM6Ly9kaWRjb21tLm9yZy9vdXQtb2YtYmFuZC8yLjAvaW52aXRhdGlvbiIsImlkIjoiNjkyMTJhM2EtZDA2OC00ZjlkLWEyZGQtNDc0MWJjYTg5YWYzIiwiZnJvbSI6ImRpZDpleGFtcGxlOmFsaWNlIiwiYm9keSI6eyJnb2FsX2NvZGUiOiIiLCJnb2FsIjoiIn0sImF0dGFjaG1lbnRzIjpbeyJpZCI6InJlcXVlc3QtMCIsIm1lZGlhX3R5cGUiOiJhcHBsaWNhdGlvbi9qc29uIiwiZGF0YSI6eyJqc29uIjoiPGpzb24gb2YgcHJvdG9jb2wgbWVzc2FnZT4ifX1dfQ
+```
```

Comment:
It is good to remind what an OOB message is but it must remain brief. In fact, what one would expect from this documentation is for example the version of the protocol implemented, the endpoints available and their purpose, etc. Instead, you seem to be giving implementation guidelines (the project have likely followed them), but this is not the right place for that.

Reply:
OK, let fix that